### PR TITLE
Tweak: Favorites widget - add an indication that a widget was added [ED-5500]

### PR DESCRIPTION
**Pull request by YitzhakAvm**

### CODE REVIEW

**Review by Nevoss — APPROVED** [review]

File: modules/favorites/assets/js/editor/types/widgets/widgets.js [hunk 1]
```diff
@@ -134,7 +134,22 @@ export default class Widgets extends FavoriteType {
...
+							this.toggle( widget );
+
+							// Replace context-menu button with "Added" after the widget became a favorite
+							if ( this.isFavorite( widget ) ) {
+								// `setTimeout` is used because for some reason when a context-menu button is clicked,
+								// the modal `hide` method is invoked by other dialog usages. Therefore, we wait till it
+								// happens and then show it again.
+								setTimeout( () => {
+									modal.show();
+									modal.updateActionTitle( { ...action, title: () => 'Added' } );
```
Comment (matipojo) [inline modules/favorites/assets/js/editor/types/widgets/widgets.js:148]: Translate

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/favorites/assets/js/editor/types/widgets/widgets.js (hunk 1 above)
Comment (ItzhakAvm) [reply modules/favorites/assets/js/editor/types/widgets/widgets.js]: Done.

File: modules/favorites/assets/js/editor/types/widgets/widgets.js [hunk 2]
```diff
@@ -134,7 +134,25 @@ export default class Widgets extends FavoriteType {
 						title: this.isFavorite( widget ) ?
 							__( 'Remove from Favorites', 'elementor' ) :
 							__( 'Add to Favorites', 'elementor' ),
-						callback: () => this.toggle( widget ),
+						callback: ( modal, action ) => {
+							// Toggle favorite state
+							this.toggle( widget );
+
+							// Replace context-menu button with "Added" after the widget became a favorite
+							if ( this.isFavorite( widget ) ) {
```
Comment (matipojo) [inline modules/favorites/assets/js/editor/types/widgets/widgets.js:142]: Please show a Toast message instead

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/favorites/controller.php [hunk 3]
```diff
@@ -35,6 +35,14 @@ public function delete_item( $request ) {
 		return $module->get( $type );
 	}
 
+	public function create_item_permissions_check( $request ) {
+		return current_user_can( 'edit_posts' );;
```
Comment (matipojo) [inline modules/favorites/controller.php:39]:
```suggestion
		return current_user_can( 'edit_posts' );
```

File: tests/playwright/assets/modules/favorites/widgets.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		const openPanelElementContextMenu = async () => {
+			await this.page.click( `.elementor-element >> :text("${ widgetTitle }")`, { button: 'right' } );
+			await this.page.waitForSelector( '.elementor-context-menu' );
+		};
+		await openPanelElementContextMenu();
+
+		if ( await this.page.$( ':text("Remove from Favorites")' ) ) {
+			await this.page.click( ':text("Remove from Favorites")' );
+			await openPanelElementContextMenu();
+		}
```
Comment (matipojo) [inline tests/playwright/assets/modules/favorites/widgets.js:18]: Why do you test in a local environment?

File: tests/playwright/config/global-setup.js [hunk 5]
```diff
@@ -7,8 +7,9 @@ module.exports = async ( config ) => {
 	await page.goto( `${ config.projects[ 0 ].use.baseURL }/wp-admin` );
 
 	await page.waitForSelector( 'text=Log In' );
-	await page.fill( 'input[name="log"]', 'admin' );
-	await page.fill( 'input[name="pwd"]', 'password' );
+	await page.fill( 'input[name="log"]', process.env.UNAME || 'admin' );
+	await page.waitForTimeout( 500 );
+	await page.fill( 'input[name="pwd"]', process.env.PASSWD || 'password' );
```
Comment (matipojo) [inline tests/playwright/config/global-setup.js:12]: It should be tested in a clean environment.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/playwright/assets/modules/favorites/widgets.js (hunk 4 above)
Comment (ItzhakAvm) [reply tests/playwright/assets/modules/favorites/widgets.js:18]: I can move to a testing environment, but it doesn't change the fact that previous test in the same run might try to add the same favorite. Therefore, it should be reset.

File: tests/playwright/config/global-setup.js (hunk 5 above)
Comment (ItzhakAvm) [reply tests/playwright/config/global-setup.js:12]: Do clean environments necessarily have the same credentials?

File: modules/favorites/controller.php (hunk 3 above)
Comment (ItzhakAvm) [reply modules/favorites/controller.php]: Fixed.

File: assets/dev/js/editor/utils/context-menu.js [hunk 6]
```diff
@@ -110,7 +110,7 @@ module.exports = elementorModules.Module.extend( {
 			return;
 		}
 
-		action.callback();
+		action.callback( this, action );
```
[the remark this answers is not in the document]
Comment (ItzhakAvm) [reply assets/dev/js/editor/utils/context-menu.js]: Forgot to remove.

File: tests/playwright/assets/modules/favorites/widgets.js (hunk 4 above)
Comment (matipojo) [reply tests/playwright/assets/modules/favorites/widgets.js:18]: The test should cleanup itself instead of clean up another test

Comment (ItzhakAvm) [reply tests/playwright/assets/modules/favorites/widgets.js]: Done.

**Review by matipojo — APPROVED** [review]

**Pull request merged**